site/src/pages/components/switch.explainer.mdx

```diff
@@ -0,0 +1,574 @@
+| `thumb`         | Enables slotting custom content into the thumb.                                          | A default thumb implementation is provided. |
+| `track`         | Enables slotting of custom elements inside the track, e.g. for more intricate animations | No default content                          |
+| `toggled` (?)   | Enables inserting content on the toggled (end) side of the track                         | No default content                          |
+| `untoggled` (?) | Enables inserting content on the non-toggled (start) side fo the track                   | No default content                          |
+
+---
+
+## Behavior
+
+### States and Interactions
+
+{/* _List key component states, valid state transitions, and how interactions trigger state transitions._ */}
+
+{/* - _Are states changed as a result of keyboard? mouse? touch?_ */}
+{/* - _Are there corresponding keyboard interactions for mouse/touch interactions?_ */}
+{/* - _Are some events triggered on key down vs key press?_ */}
+
+<div class="question">
+  Should the switch value type be a boolean or the string set ["on", "off"]?
+</div>
+
+| State Group | States            | Initial State | Description                                                                                                           | Interaction/Transition                                                                                      |
+| ----------- | ----------------- | ------------- | --------------------------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------- |
+| `value`     | `"on"`/`"off"`(?) | `"off"`       | Determines the current value of the component.                                                                        | Changed upon user interaction on click, touch, space key (when focused), enter key (when focused, optional) |
+| `disabled`  | `true`/`false`    | `false`       | When `true`, disables the control, preventing user interaction and displaying the control with a disabled appearance. | No interaction. Controlled programmatically.                                                                |
+| `focused`   | `true`/`false`    | `false`       |                                                                                                                       |                                                                                                             |
+
+### Accessibility
+
+The control can be in a checked or unchecked state. [^6]
+
+The `<switch>` element can be assigned the default role of switch as described in the [Core Accessibility API Mappings](https://w3c.github.io/core-aam/#role-map-switch). [^7]
+
+Content for thumb and either track side can be defined by the author.
```

Comment:
since this is the accessibility section, it's important to note that the content defined in the switch will not likely be exposed without rethinking how this content can be mapped.  Yet another reason, aside from the potential for the awkward sizing of the switch, why i think it's a bad idea to put text content within the switch parts.

The switch should be named by its label (so this should also be calling out it would need to be a labellable element).  The content of the swtich, if any, really should just be on/off/a localized version of these values - and then it can just be safely ignored since the checked state will expose it as either being 'on' or 'off' via the a11y api.

Reply:
I opened https://github.com/openui/open-ui/issues/959 to gather some ideas or solutions. The more I think about it, the more I think it's not a good idea to add content, except for images. A checkmark on an "on" switch could help communicate that it's "on" without just using color while avoiding most of the issues. I'm just not sure how that could be implemented.

Reply:
Resolved on #979 and #978.